beacon-chain/sync/initial-sync/blocks_fetcher_peers.go

```diff
@@ -118,3 +120,39 @@ func (f *blocksFetcher) filterPeers(peers []peer.ID, peersPercentage float64) ([
 
 	return peers, nil
 }
+
+// filterScoredPeers returns transformed list of peers,
+// weight sorted by scores and capacity remaining.
+func (f *blocksFetcher) filterScoredPeers(ctx context.Context, peers []peer.ID, ratio float64) ([]peer.ID, error) {
+	ctx, span := trace.StartSpan(ctx, "initialsync.filterScoredPeers")
+	defer span.End()
+
+	if len(peers) == 0 {
+		return peers, nil
+	}
+
+	// Sort peers using both block provider score and, custom, capacity based score (see
+	// peerFilterCapacityWeight if you want to give different weights to provider's and capacity
+	// scores).
+	// Scores produced are used as weights, so peers are ordered probabilistically i.e. peer with
+	// a higher score has higher chance to end up higher in the list.
+	scorer := f.p2p.Peers().Scorers().BlockProviderScorer()
+	peers = scorer.WeightSorted(f.rand, peers, func(peerID peer.ID, blockProviderScore float64) float64 {
+		remaining, capacity := float64(f.rateLimiter.Remaining(peerID.String())), float64(f.rateLimiter.Capacity())
+		// When no capacity for a good peer left, allow less performant peer to take a chance.
+		if remaining < float64(f.blocksPerSecond) {
```

Comment:
For this to be true wouldnt `remaining` be equal to zero ? 
if `remaining` <  `f.blocksPerSecond` , there is still some capacity, no ?

Reply:
Well, I suppose this is an issue in comment, it should read as "When remaining capacity approaches to exhaustion.."